src/zarr.cpp

```diff
@@ -587,6 +599,24 @@ zarr::Zarr::set_chunking(const ChunkingProps& props, const ChunkingMeta& meta)
     CHECK(planes_per_chunk_ > 0);
 }
 
+void
+zarr::Zarr::reset_state_()
+{
+    // reset state variables
+    writers_.clear();
+
+    // should be empty, but just in case
+    for (auto& [_, frame] : scaled_frames_) {
+        if (frame.has_value() && frame.value()) {
+            free(frame.value());
+        }
+    }
+    scaled_frames_.clear();
+
+    error_ = false;
+    error_msg_.clear();
```

Comment:
`set_error` (just below) acquires a lock on `mutex_`, but this does not. Is that intended? There are other places where `error_` is mutated and the lock is not held, so this inconsistency may not be new. But it would be good to not add to it.

Reply:
Yeah, by this point the thread pool is shut down, destroyed even. Nobody else should be able to mutate it.